ehrapy/api/preprocessing/encoding/_encode.py

```diff
@@ -504,6 +594,99 @@ def _undo_encoding(
             layers={"original": new_x.copy()},
         )
 
+    @staticmethod
+    def _delete_all_encodings(adata: AnnData) -> Tuple[Optional[np.ndarray], Optional[list]]:
+        """Delete all encoded columns and keep track of their indices
+        Args:
+            adata: The AnnData object to operate on
+
+        Returns:
+            A temporary X were all encoded columns are deleted and all var_names of unencoded columns
+        """
+        var_names = list(adata.var_names)
+        if adata.X is not None and var_names is not None:
+            idx = 0
+            for var_name in var_names:
+                if not var_name.startswith("ehrapycat"):
+                    break
+                idx += 1
+            # case: only encoded columns were found
+            if idx == len(var_names):
+                return None, None
+            # don't need to consider case when no encoded columns are there, since undo_encoding would not run anyways in this case
+            return adata.X[:, idx:].copy(), var_names[idx:]
+        return None, None
+
+    @staticmethod
+    def _reorder_encodings(adata: AnnData, new_encodings: Dict[str, Union[List[List[str]], List[str]]]):
+        """Reorder the encodings and update which column will be encoded using which mode (with which columns in case of
+        multi column encoding modes).
+        Args:
+            adata: The AnnData object to be reencoded
+            new_encodings: The new encodings passed by the user (might affect encoded as well as previously non encoded columns)
+
+        Returns:
+            An updated encoding scheme
+        """
+        flattened_modes: List[Union[List[str], str]] = sum(new_encodings.values(), [])  # type: ignore
```

Comment:
Why type ignore here?

Reply:
Mypy complained about a missing type annotation. Did not really get why tough

Reply:
Time to figure it out :)

Reply:
Guess it was due to `sum` expecting `Iterable[]` but the values are like `dict_values`. But it is one of the recommended ways to do this, so I guess it's fine?